src/streaming/extensions/TextTrackExtensions.js

```diff
@@ -30,28 +30,66 @@
  */
 MediaPlayer.utils.TextTrackExtensions = function () {
     "use strict";
-    var Cue;
+    var Cue,
+        textTrackQueue = [],
+        trackElementArr = [],
+        currentTrackIdx = 0;
 
     return {
+        mediaController:undefined,
+
         setup: function() {
             Cue = window.VTTCue || window.TextTrackCue;
         },
 
-        addTextTrack: function(video, captionData,  label, scrlang, isDefaultTrack) {
-
-            //TODO: Ability to define the KIND in the MPD - ie subtitle vs caption....
-            this.track = video.addTextTrack("captions", label, scrlang);
-            // track.default is an object property identifier that is a reserved word
-            // The following jshint directive is used to suppressed the warning "Expected an identifier and instead saw 'default' (a reserved word)"
-            /*jshint -W024 */
-            this.track.default = isDefaultTrack;
-            this.track.mode = "showing";
-            this.video=video;
-            this.addCaptions(0, captionData);
-            return this.track;
+        addTextTrack: function(textTrackInfoVO, totalTextTracks) {
+
+            textTrackQueue.push(textTrackInfoVO);
+            if (this.video === undefined) {
+                this.video = textTrackInfoVO.video;
+            }
+
+            if(textTrackQueue.length === totalTextTracks) {
+
+                if (totalTextTracks > 1) { // sort multi text tracks alphabetically. Not sure how we to sort tracks and flag a track as default.  Currently sort alph, and default track is first media item in MPD.
+                    textTrackQueue.sort(function(a,b) {
+                        return a.lang > b.lang;
+                    });
+                }
+
+                var defaultIndex = 0;
+                for(var i = 0 ; i < textTrackQueue.length; i++) {
+                    var track = document.createElement('track'),//this.getCurrentTextTrack(),
+                        captionType = textTrackQueue[i].role ? textTrackQueue[i].role : "captions";
+
+                    currentTrackIdx = i;
+                    trackElementArr.push(track);
+
+                    track.kind = captionType === "subtitle" ? "subtitles" : captionType;
+                    track.label = textTrackQueue[i].lang;
+                    track.srclang = textTrackQueue[i].lang;
+
+                    // track.default is an object property identifier that is a reserved word
+                    // The following jshint directive is used to suppressed the warning "Expected an identifier and instead saw 'default' (a reserved word)"
+                    /*jshint -W024 */
+                    track.default = textTrackQueue[i].defaultTrack;
+                    if (textTrackQueue[i].defaultTrack) {
+                        defaultIndex = i;
+                    }
+
+                    this.video.appendChild(track);
```

Comment:
Ah the critical questions.   https://github.com/videojs/video.js/issues/2369 
So with addTextTrack API the removeTextTrack API has been deprecated for unknown reasons so there is no way to remove tracks once added.  You can just set disabled and this causes issues.   Me and the Video.js guys agreed that we should handle this manually.   Check out how nice delete tracks are now compared to the hacking that was done before.    This method is not without issues,   which we are trying to solve currently in that thread.   This includes being able to se a certain track as default .